tests/manage/pv_services/test_overprovision_level_policy_control.py

```diff
@@ -0,0 +1,233 @@
+@pytest.mark.polarion_id("OCS-4472")
+class TestOverProvisionLevelPolicyControl(ManageTest):
+    """
+    Test OverProvision Level Policy Control
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            log.info("Delete overprovisionControl from storage cluster yaml file")
+            storagecluster_obj = OCP(
+                resource_name=constants.DEFAULT_CLUSTERNAME,
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                kind=constants.STORAGECLUSTER,
+            )
+            params = '{"spec": {"overprovisionControl": []}}'
+            storagecluster_obj.patch(
+                params=params,
+                format_type="merge",
+            )
+            log.info("Verify storagecluster on Ready state")
+            verify_storage_cluster()
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.parametrize(
+        argnames=["sc_name", "sc_type"],
+        argvalues=[
+            pytest.param(*[constants.CEPHBLOCKPOOL_SC, constants.CEPHBLOCKPOOL]),
+            pytest.param(*[constants.CEPHFILESYSTEM_SC, constants.CEPHFILESYSTEM]),
+            pytest.param(*["sc-test-blk", constants.CEPHBLOCKPOOL]),
+            pytest.param(*["sc-test-fs", constants.CEPHFILESYSTEM]),
+        ],
+    )
+    def test_over_provision_level_policy_control(
+        self,
+        setup_sc,
+        sc_name,
+        sc_type,
+        teardown_project_factory,
+        pvc_factory,
+        pod_factory,
+        storageclass_factory,
+    ):
+        """
+        Test Process:
+            1.Create project with “openshift-quota” label
+            2.Create new Storage Class or use ceph-rbd or ceph-fs
+            3.Add “overprovisionControl” section to storagecluster yaml file [max=8Gi]
+            4.Verify storagecluster on Ready state
+            5.Create 5Gi pvc on project “ocs-quota-sc-test” with sc “sc-test”
+            6.Create new pvc with 6Gi capacity and verify it failed [6Gi + 5Gi > 8Gi]
+            7.Resize PVC to 20Gi and verify it failed [20Gi > 8Gi]
+            8.Resize the PVC to 6Gi and verify it is working [8Gi > 6Gi]
+            9.Create New PVC with 1G capacity and verify it is working [8Gi > 1Gi + 6Gi]
+
+        """
+        quota_name = {
+            constants.CEPHBLOCKPOOL_SC: "ocs-storagecluster-ceph-rbd-quota-sc-test",
+            constants.CEPHFILESYSTEM_SC: "ocs-storagecluster-cephfs-quota-sc-test",
+            "sc-test-blk": "sc-test-blk-quota-sc-test",
+            "sc-test-fs": "sc-test-fs-quota-sc-test",
+        }
+
+        log.info("Create project with “openshift-quota” label")
+        ns_quota = templating.load_yaml(constants.NAMESPACE_QUOTA)
+        ns_quota_obj = OCS(**ns_quota)
+        ocs_project_obj = ns_quota_obj.create()
+        ocp_project_obj = OCP(
+            kind="Project", namespace=ocs_project_obj["metadata"]["name"]
+        )
+        teardown_project_factory(ocp_project_obj)
```

Comment:
Lines 102-108 can be replace by functions :
ocp.new_project()
ocp.add_lable()

Reply:
done